data/trie/trieStorageManager.go

```diff
@@ -177,30 +177,32 @@ func (tsm *trieStorageManager) Database() data.DBWriteCacher {
 	return tsm.db
 }
 
-// EnterSnapshotMode sets the snapshot mode on
-func (tsm *trieStorageManager) EnterSnapshotMode() {
+// EnterPruningBufferingMode increases the counter that tracks how many operations
+// that block the pruning process are in progress
+func (tsm *trieStorageManager) EnterPruningBufferingMode() {
 	tsm.storageOperationMutex.Lock()
 	defer tsm.storageOperationMutex.Unlock()
 
-	tsm.snapshotInProgress++
+	tsm.pruningBlockingOps++
 
-	log.Trace("enter snapshot mode", "snapshots in progress", tsm.snapshotInProgress)
+	log.Trace("enter pruning buffering state", "operations in progress that block pruning", tsm.pruningBlockingOps)
 }
 
-// ExitSnapshotMode sets the snapshot mode off
-func (tsm *trieStorageManager) ExitSnapshotMode() {
+// ExitPruningBufferingMode decreases the counter that tracks how many operations
+// that block the pruning process are in progress
+func (tsm *trieStorageManager) ExitPruningBufferingMode() {
 	tsm.storageOperationMutex.Lock()
 	defer tsm.storageOperationMutex.Unlock()
 
-	if tsm.snapshotInProgress < 1 {
-		log.Error("ExitSnapshotMode called too many times")
+	if tsm.pruningBlockingOps < 1 {
+		log.Error("ExitPruningBufferingMode called too many times")
```

Comment:
we can return here and remove the if on L201

Reply:
Done.